apps/emqx_connector/src/emqx_connector_mongo.erl

```diff
@@ -315,47 +318,52 @@ may_parse_srv_and_txt_records_(#{mongo_type := Type,
         true ->
             error({missing_parameter, replica_set_name});
         false ->
-            Config#{hosts => parse_servers(Type, Servers)}
+            Config#{hosts => servers_to_bin(Servers)}
     end;
 may_parse_srv_and_txt_records_(#{mongo_type := Type,
                                  srv_record := true,
                                  servers := Servers} = Config) ->
-    NServers = binary_to_list(Servers),
-    Hosts = parse_srv_records(Type, NServers),
-    ExtraOpts = parse_txt_records(Type, NServers),
+    Hosts = parse_srv_records(Type, Servers),
+    ExtraOpts = parse_txt_records(Type, Servers),
     maps:merge(Config#{hosts => Hosts}, ExtraOpts).
 
-parse_srv_records(Type, Server) ->
-    case inet_res:lookup("_mongodb._tcp." ++ Server, in, srv) of
-        [] ->
-            error(service_not_found);
-        Services ->
-            case [Host ++ ":" ++ integer_to_list(Port) || {_, _, Port, Host} <- Services] of
-                [H | _] when Type =:= single ->
-                    [H];
-                Hosts ->
-                    Hosts
-            end
+parse_srv_records(Type, Servers) ->
+    Fun = fun(AccIn, {IpOrHost, _Port}) ->
+                  case inet_res:lookup("_mongodb._tcp." ++ ip_or_host_to_string(IpOrHost), in, srv) of
+                      [] ->
+                          error(service_not_found);
+                      Services ->
+                          [ [server_to_bin({Host, Port}) || {_, _, Port, Host} <- Services]
+                          | AccIn]
+                  end
+          end,
+    Res = lists:foldl(Fun, [], Servers),
+    case Type of
+        single -> lists:nth(1, Res);
```

Comment:
Do we need a warning log? if the Res length > 1

Reply:
Log with srv record `priority` and `weight` parse in next PR.